### ISSUE TITLE
[🚨 Bug] Text-to-Audio pipeline is broken for Speech T5 TTS

### ISSUE THREAD

**Issue by Vaibhavs10** [huggingface/transformers#26401]
### System Info

- `transformers` version: 4.33.2
- Platform: Linux-5.15.120+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.17.2
- Safetensors version: 0.3.3
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu118 (False)
- Tensorflow version (GPU?): 2.13.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.7.2 (cpu)
- Jax version: 0.4.14
- JaxLib version: 0.4.14
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@sanchit-gandhi @ylacombe

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Speech T5 TTS expects speaker embeddings to be passed for understandable speech to be produced. At the moment in the `Text-to-Audio` pipeline we just call the model (processor) + generate without any speaker embeddings. This results in garbage audio as an output.

Full-repro here: https://github.com/Vaibhavs10/scratchpad/blob/main/transformers_speecht5_tts_repro.ipynb

### Expected behavior

Text-to-Audio pipeline for SpeechT5 TTS checkpoints outputs audio that is understandable.

**Comment by Vaibhavs10** [comment]
cc: @Narsil for vis.

**Comment by Narsil** [comment]
@ylacombe Maybe ?

**Comment by ylacombe** [comment]
Hey @Narsil and @Vaibhavs10,
At the moment, you have to pass the speaker embeddings as a forward argument:

```python
speaker_embeddings = torch.tensor(embeddings_dataset[7306]["xvector"]).unsqueeze(0)
forward_params = {
    "speaker_embeddings": speaker_embeddings,
}
speech = speecht5_tts("Hello, my dog is cute!", forward_params = forward_params)
```

For reference, see [this snippet](https://huggingface.co/suno/bark-small/discussions/4#64f5c7c11c2c6f58c91c8ad1) for Bark.

Adding a default speechT5 speaker embedding in the pipeline would imply model-specific code inside the pipeline code, which is [something](https://github.com/huggingface/transformers/pull/24952/#discussion_r1280336582) we'd like to avoid rn

Happy to hear about your thoughts on this, though.

**Comment by ylacombe** [comment]
I agree @Vaibhavs10! But instead of adding this to the pipeline, let's add this to [_generate_speech](https://github.com/huggingface/transformers/blob/6da93f5580e109fad5f7b523cf2b6e8a5bafb623/src/transformers/models/speecht5/modeling_speecht5.py#L2532) from `modeling_speecht5.py` since it is an issue specifically related to the model.

I'll make a PR this afternoon if that's okay with you
